lib/charms/layer/apache_kafka.py

```diff
@@ -14,14 +16,29 @@ def __init__(self, dist_config=None):
         self.resources = {
             'kafka': 'kafka-%s' % utils.cpu_arch(),
         }
-        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def verify_resources(self):
```

Comment:
One thing I did here in our own charms was add a filesize threshold so we could "work around" the zerobyte resource we're publishing our charms with. I suggest you add an additional check:

```
import os
if os.stat(self.resources['kafka']).st_size < 100000:
    return False
```

or something to this effect. this makes the resource validation a bit more robust (must be > 1mb to be considered a valid resource)